runner/src/main.rs

```diff
@@ -667,25 +655,52 @@ fn run_example(opt: &RunExamples, example: &Example) -> Step {
     // clients in the foreground.
     #[allow(clippy::collapsible_if)]
     let run_backend_server_clients: Step = if opt.run_server.unwrap_or(true) {
-        let run_server_clients = if opt.build_client.client_variant != NO_CLIENTS {
-            Step::WithBackground {
-                name: "background server".to_string(),
-                background: run_server,
-                foreground: Box::new(run_clients),
+        let run_server_clients = if example.applications.is_empty() {
+            if opt.build_client.client_variant != NO_CLIENTS {
```

Comment:
Can you flip this condition, in order to avoid keeping track of the double negation in the else branch?